File: android/app/src/main/kotlin/uz/vizara/tcall/network/ApiModels.kt
package uz.vizara.tcall.network

import com.google.gson.annotations.SerializedName

data class UserDto(
    @SerializedName("userId") val userId: String,
    @SerializedName("email") val email: String,
    @SerializedName("name") val name: String,
    @SerializedName("language") val language: String,
    @SerializedName("tcallId") val tcallId: String,
    @SerializedName("translationMode") val translationMode: String? = "text",
)

data class LoginRequest(
    val email: String,
    val password: String,
    val remember: Boolean = true,
)

data class LoginResponse(
    val user: UserDto?,
    val token: String? = null,
    val error: String? = null,
)

data class SessionResponse(
    val user: UserDto?,
)

File: platforms/ios/Tcall/Models/ApiModels.swift
import Foundation

enum TcallConfig {
    static let apiBaseURL = URL(string: "https://api.tcall.uz")!
    static let appVersion = "1.000000"
    static let nativeHeader = "X-Tcall-Native"
    static let nativeHeaderValue = "1"
}

struct UserDto: Codable {
    let userId: String
    let email: String
    let name: String
    let language: String
    let tcallId: String
}

struct LoginRequest: Codable {
    let email: String
    let password: String
    let remember: Bool
}

struct LoginResponse: Codable {
    let user: UserDto?
    let token: String?
    let error: String?
}

struct SessionResponse: Codable {
    let user: UserDto?
}
